src/main/java/org/jabref/logic/exporter/TemplateExporter.java

```diff
@@ -257,7 +277,11 @@ public void export(final BibDatabaseContext databaseContext, final Path file,
 
                 // Write the entry
                 if (layout != null) {
-                    ps.write(layout.doLayout(entry, databaseContext.getDatabase()));
+                    if (deleteBlankLines) {
+                        ps.write(layout.doLayout(entry, databaseContext.getDatabase()).replaceAll("(?m)^\\s", ""));
```

Comment:
Please extract the regex to Pattern. Compile. You  can then use it with
.matcher(structure).replaceAll(replace) 
And I would suggest a small comment explaining the regex

Reply:
No problem, done! Although I hardly suspect that this could become a performance problem due to frequent compilation. After all, this would be executed once per RIS entry and even a few thousand compilations probably won't make a noticeable difference.

But well, it surely doesn't hurt to compile only once.